ui/details-panel/autoclassify/LineOption.jsx

```diff
@@ -0,0 +1,206 @@
+       controller: 'BugFilerCtrl',
+       size: 'lg',
+       openedClass: 'filer-open',
+       resolve: {
+         summary: () => errorLine.data.bug_suggestions.search,
+         search_terms: () => errorLine.data.bug_suggestions.search_terms,
+         fullLog: () => logUrl,
+         parsedLog: () => `${location.origin}/${getLogViewerUrl(job.id, repoName)}`,
+         reftest: () => (isReftest(job) ? `${reftestUrl}&only_show_unexpected=1` : ''),
+         selectedJob: () => job,
+         allFailures: () => [errorLine.data.bug_suggestions.search.split(" | ")],
+         crashSignatures: () => crashSignatures,
+         successCallback: () => (data) => {
+           const bugId = data.success;
+           window.open(getBugUrl(bugId));
+           onManualBugNumberChange(optionModel, `${bugId}`);
+         }
+       }
+     });
+
+    selectedOption.id = optionModel.id;
+    modalInstance.opened.then(() => modalInstance.initiate());
+  }
+
+  render() {
+    const {
+      job,
+      errorLine,
+      optionModel,
+      selectedOption,
+      canClassify,
+      pinBoard,
+      onOptionChange,
+      onIgnoreAlwaysChange,
+      ignoreAlways,
+      manualBugNumber,
+      onManualBugNumberChange,
+    } = this.props;
+    const option = optionModel;
+    const bugSummary = { __html: highlightCommonTerms(
+      option.bugSummary, errorLine.data.bug_suggestions.search) };
+
+    return (
+      <div className="classification-option">
+        <span className="classification-icon">
+          {option.isBest ?
+            <span className="fa fa-star-o" title="Autoclassifier best match" /> :
+            <span className="classification-no-icon">&nbsp;</span>}
+        </span>
+
+        <FormGroup check>
+          <Label check>
+            {!(option.type === 'classifiedFailure' && !option.bugNumber) && <Input
+              type="radio"
+              value={option}
+              id={option.id}
+              checked={selectedOption.id === option.id}
+              name={errorLine.id}
+              onChange={() => onOptionChange(option)}
+              className={canClassify ? '' : 'hidden'}
+            />}
+            {!!option.bugNumber && <span className="line-option-text">
+              {(!canClassify || pinBoard.isPinned(job)) &&
+                <button
+                  className="btn btn-xs btn-light-bordered"
+                  onClick={() => pinBoard.addBug({ id: option.bugNumber }, job)}
+                  title="add to list of bugs to associate with all pinned jobs"
+                ><i className="fa fa-thumb-tack" /></button>}
+              {!!option.bugResolution &&
+                <span className="classification-bug-resolution"> [{option.bugResolution}] </span>}
+              <a
+                href={getBugUrl(option.bugNumber)}
+                target="_blank"
+                rel="noopener"
+              >{option.bugNumber} -
+                <span dangerouslySetInnerHTML={bugSummary} />
```

Comment:
I don't think we should add any more usages of `dangerouslySetInnerHTML` - could you adjust this and the others? :-)

Reply:
I think React should cope better with this use-case than AngularJS did, making the need to manually escape and use `dangerouslySetInnerHTML` redundant?